scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,607 @@
+##############################################################################
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
```

Comment:
@andyfaff that doesn't sound quite right to me. As @sgubianpm is the author of the R code as well, he's free to relicense that code or any derived code under any license of his choosing. You can also dual license code as GPL and something else, so that FAQ entry doesn't really apply here.